scripts/utils/internals.js

```diff
@@ -100,6 +100,14 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     }
   }
 
+  function signHashWithPrivateKey(hash, privateKey) {
+    const Util = require("ethereumjs-util")
+
+    const msgBuff = new Buffer(Util.stripHexPrefix(hash), "hex")
+    const sig = Util.ecsign(msgBuff, new Buffer(privateKey, "hex"))
+    return "0x" + sig.r.toString("hex") + sig.s.toString("hex") + sig.v.toString(16)
```

Comment:
Is the padding right here? Unless `sig.r`, `sig.s`, `sig.v`, have an ad hoc implementation of `toString` that takes into account padding, I believe the returned hash would be incorrect if any of these parameters started with a zero byte.

Reply:
We are code already in the function:
```
  function signTransaction(lw, signers, transactionHash) 
```

Reply:
```
const buf1 = Buffer.allocUnsafe(26);

buf1[0] = 0 ;
for (let i = 20; i < 26; i++) {
  // 97 is the decimal ASCII value for 'a'.
  buf1[i] = i ;
}
console.log(buf1);
console.log(buf1.toString('hex'));
```
use this small script to convince yourself that buffers do not cut of paddings

Reply:
I didn't notice we already use this code in another function in our code. This is indeed copied from the Gnosis Safe code, so even if the padding was incorrect this would be a "feature" rather than a bug. It would be nicer if we could import all of this from their repo instead, but after a brief look it doesn't look like this is possible.
(By the way, an implicit assumption of your example is that the two buffers always have the same length for different signatures, which might not be the case. Also, the last output is a JS number, so this is intentionally not padded, which to me is a bit surprising.)